src/utils/chess.ts

```diff
@@ -76,15 +76,30 @@ const FEN_PIECE_TYPE_MAP: { [key: string]: PieceType } = {
   q: "queen",
   k: "king",
 };
-const REVERSE_FEN_PIECE_TYPE_MAP: Record<PieceType, string> = Object.keys(
-  FEN_PIECE_TYPE_MAP
-).reduce(
-  (acc, key) => {
-    acc[FEN_PIECE_TYPE_MAP[key]] = key;
-    return acc;
-  },
-  {} as Record<PieceType, string>
-);
+export const REVERSE_FEN_PIECE_TYPE_MAP: Record<PieceType, string> =
+  Object.keys(FEN_PIECE_TYPE_MAP).reduce(
+    (acc, key) => {
+      acc[FEN_PIECE_TYPE_MAP[key]] = key;
+      return acc;
+    },
+    {} as Record<PieceType, string>
+  );
+
+export function addCustomPieceTypes(map: Record<string, string>): void {
```

Comment:
Is this function idempotent? What happens if I call it twice? I think this will add duplicate values to `PIECE_TYPES` right? Can we prevent against that?